go/internal/controllers/stress_controller.go

```diff
@@ -0,0 +1,58 @@
+package controllers
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/scylladb/gocqlx/v2"
+)
+
+type StressController struct {
+	Session *gocqlx.Session
+}
+
+func NewStressController(session *gocqlx.Session) *StressController {
+	return &StressController{Session: session}
+}
+
+func (c *StressController) Stress() error {
+	fmt.Println("------------------------------------")
+	fmt.Println("Inserting 100,000 records into the database...")
+	fmt.Println("> Starting...")
+
+	start := time.Now()
+
+	var wg sync.WaitGroup
+	sem := make(chan bool, 550)
+
+	for i := 0; i < 100_000; i++ {
+		sem <- true 
```

Comment:
I'm not sure what's the purpose of this channel. Since it will never block the defered function from go-routine (since you're already adding a value to it)

Reply:
The channel will block as soon as it reaches its maximum capacity (550), preventing new go-routines from being created until the space is freed up.